src/main/java/org/gridsuite/modification/server/modifications/ByFormulaModification.java

```diff
@@ -0,0 +1,187 @@
+        filterService = modificationApplicator.getFilterService();
+    }
+
+    @Override
+    public void check(Network network) throws NetworkModificationException {
+        if (modificationInfos == null) {
+            throw new NetworkModificationException(NetworkModificationException.Type.BY_FORMULA_MODIFICATION_ERROR, "Missing required attributes to modify the equipment");
+        }
+
+        if (CollectionUtils.isEmpty(modificationInfos.getFormulaInfosList())) {
+            throw new NetworkModificationException(NetworkModificationException.Type.BY_FORMULA_MODIFICATION_ERROR, "At least one formula is required");
+        }
+
+        if (modificationInfos.getFormulaInfosList().stream().anyMatch(formulaInfos -> CollectionUtils.isEmpty(formulaInfos.getFilters()))) {
+            throw new NetworkModificationException(NetworkModificationException.Type.BY_FORMULA_MODIFICATION_ERROR, "Every formula must have at least one filter");
+        }
+    }
+
+    @Override
+    public void apply(Network network, Reporter subReporter) {
+        // collect all filters from all variations
+        var filters = modificationInfos.getFormulaInfosList().stream()
+                .flatMap(v -> v.getFilters().stream())
+                .filter(distinctByKey(FilterInfos::getId))
+                .collect(Collectors.toMap(FilterInfos::getId, FilterInfos::getName));
+
+        Map<UUID, FilterEquipments> exportFilters = getUuidFilterEquipmentsMap(network, subReporter, filters);
+
+        if (exportFilters != null) {
+            Reporter formulaSubReporter = subReporter.createSubReporter("appliedFormulasModifications", "Formulas");
+            List<Report> formulaReports = new ArrayList<>();
+            modificationInfos.getFormulaInfosList().forEach(formulaInfos ->
+                    formulaInfos.getFilters().forEach(filterInfos ->
+                            applyFormulaOnFilterEquipments(network, exportFilters, formulaReports, formulaInfos, filterInfos)));
+
+            createReport(subReporter, "byFormulaModification", "new modification by formula", TypedValue.INFO_SEVERITY);
+            formulaSubReporter.report(Report.builder()
+                    .withKey("appliedFormulasModifications")
+                    .withDefaultMessage("  Formulas")
+                    .withSeverity(TypedValue.INFO_SEVERITY)
+                    .build());
+            formulaReports.forEach(formulaSubReporter::report);
+        }
+    }
+
+    private void applyFormulaOnFilterEquipments(Network network,
+                                                Map<UUID, FilterEquipments> exportFilters,
+                                                List<Report> formulaReports,
+                                                FormulaInfos formulaInfos,
+                                                FilterInfos filterInfos) {
+        var filterEquipments = exportFilters.get(filterInfos.getId());
```

Comment:
To enhance code readability and maintainability, I suggest specifying the explicit return type instead of using var 
```suggestion
         FilterEquipments filterEquipments = exportFilters.get(filterInfos.getId());
```

Reply:
fixed